source/common/res/features/spare-change/main.js

```diff
@@ -0,0 +1,187 @@
+(function poll() {
+  if (typeof ynabToolKit !== 'undefined' && ynabToolKit.pageReady === true) {
+    ynabToolKit.spareChange = (function () {
+      var selectedTransactions;
+      var currentlyRunning = false;
+
+      function setSelectedTransactions() {
+        var accountController = ynabToolKit.shared.containerLookup('controller:accounts');
+        var visibleTransactionDisplayItems = accountController.get('visibleTransactionDisplayItems');
+        selectedTransactions = visibleTransactionDisplayItems.filter(i => i.isChecked && i.get('outflow'));
+      }
+
+      function getSelectedAccount() {
+        var applicationController = ynabToolKit.shared.containerLookup('controller:application');
+        var selectedAccountId = applicationController.get('selectedAccountId');
+
+        if (selectedAccountId) {
+          var accountController = ynabToolKit.shared.containerLookup('controller:accounts');
+          var selectedAccounts = accountController.get('activeAccounts');
+          var selectedAccount = selectedAccounts.find(a => a.itemId === selectedAccountId);
+
+          return selectedAccount;
+        }
+
+        return null;
+      }
+
+      function updateSpareChangeCalculation() {
+        var i;
+        var transaction;
+        var runningSpareChange = 0;
+
+        var selectedAccount = getSelectedAccount();
+        if (selectedAccount) {
+          for (i = 0; i < selectedTransactions.length; i++) {
+            transaction = selectedTransactions[i];
+
+            if (transaction.get('parentEntityId') !== null) {
+              continue;
+            }
+
+            if (transaction.get('outflow')) {
+              var amount = transaction.get('outflow');
+              runningSpareChange += Math.round((Math.ceil(amount / 1000.0) - amount / 1000.0) * 1000);
```

Comment:
Definitely not going to worry about this now, but in case you didn't know, the `ynab` object has methods for converting to and from millicents already.